### ISSUE TITLE
Ignored errors can clash across different Figma files if nodes happen to share the same ID

### ISSUE THREAD

**Issue by alexjlockwood** [destefanis/design-lint#15]
Currently ignored lint errors are identified by the `node.id` associated with it. However, node IDs are only guaranteed to be unique within the current Figma file, which can cause problems.

For example, say you are in one Figma file and you ignore an error on a `TextNode` with a string ID `1:2`. Then you jump into a second Figma file, and you run lint on a `TextNode` that just happens to also have the same lint error as well as the same ID `1:2`.

Because the ignored errors are stored in local storage, the lint error in the 2nd file will not display because a very similar case just happened to be ignored in the first file.

I think maybe one possible way to address this issue is to store ignored errors on the nodes themselves using [`setPluginData()`](https://www.figma.com/plugin-docs/api/TextNode/#setplugindata). (Just an idea... haven't totally thought it through yet but seemed like it might be promising).

**Comment by destefanis** [comment]
Oh this is interesting! I'll try and think of a solution as well. Perhaps another solution is to grab some sort of data about the file itself and use that in local storage. So if you ignore one file in project A, when you open project B it won't have the same storage.

**Comment by alexjlockwood** [comment]
Yeah, I was kind of thinking about that too. Unfortunately there isn't much unique information that is available about the file using the Figma plugin API. Really the only useful information you can obtain is the file's name, and that can obviously change.

One other idea I was thinking of though that would theoretically work: generate a [unique ID](https://gist.github.com/gordonbrander/2230317) when the plugin launches for the very first time inside a Figma file. Then set it on the `figma.root` document node using `DocumentNode#setPluginData()`. On subsequent plugin launches, you can store this unique ID as a field inside the ignored error objects. When retrieving the ignored errors from local storage, you can filter out the errors that don't have a unique ID that is equal to the `DocumentNode`'s unique ID.

I'm still not totally sure if this would cover all of the edge cases though. For example, I'm not sure if node IDs are guaranteed to be unique across the entire version history of a Figma file. For example, it might be possible for (1) you create a node with ID `1:2`, (2) you ignore an error associated with this node, (3) you delete the node, and (4) you create a completely different node with the same ID `1:2`. I haven't confirmed whether this is actually possible or not... but if it is, then I guess the only solution might be to just associate ignored errors with the nodes using `setPluginData()` as I mentioned in my bug report.

It's worth mentioning that this is not a high priority, serious issue. It's an edge case for sure, and I'm not sure how often these issues would actually occur in the real world. :D

**Comment by alexjlockwood** [comment]
I ended up implementing a fix in my fork using the approach I described above... with a small tweak. Basically I create the unique ID, attach it to the `DocumentNode`, and then I store a list of ignored errors on a per-file basis (i.e. I store the lists separately in local storage by prefixing the storage key I pass to `figma.storage.{get,set}Async()` with the unique ID... i.e. `ignoredErrors` becomes `${uniqueId}_ignoredErrors`). Seems to work for now at least. 🤷‍♂
